### PR TITLE
Remove API docs from repo and autogenerate; remove hard-coded mocks

### PR DESCRIPTION
**Pull request by bnaul**
- The API doc code is now in `doc/conf.py` so that the Sphinx build generates them automatically. A little hacky, but removes the need to keep them in the repo and remember to rebuild them every time.
- Mocking is now handled by monkey patching `__import__` in the Sphinx build to use mocks where needed. No longer necessary to add new requirements to the mock list.
- The list of modules to include in the API docs is still hard-coded; I played around with auto-generating it but it was not nearly as easy to adapt the `scikit-image` code as I'd hoped, so I think this is fine for now.

Drone is happy but I can't 100% know that this works until it's merged and built on readthedocs, so if you guys aren't around I may just self-merge :smiling_imp:

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -308,21 +308,40 @@
...
-    stats = MagicMock()
-MOCK_MODULES = ['requests', 'parse', 'yaml', 'numpy', 'xray', 'sklearn',
-                'pandas', 'scipy', 'scipy.stats', 'scipy.linalg',
-                'scipy.special', 'sklearn.externals', 'sklearn.base',
-                'sklearn.ensemble', 'sklearn.linear_model', 'celery', 'dask',
-                'dask.async', 'mltsp.science_features._lomb_scargle']
-for m in MOCK_MODULES:
-    sys.modules[m] = Mock()
-sys.path.append(os.path.join(os.path.dirname(__name__), '..'))
+    import unittest.mock as mock
```
Comment (acrellin) [inline doc/conf.py]: Is it just me or is the indentation crazy here?

**Comment by acrellin** [comment]
The changes look good, but drone doesn't look very happy to me :disappointed:

**Comment by stefanv** [comment]
I wish I knew why Drone kept reporting success without completion.

**Comment by bnaul** [comment]
Am I in a parallel universe? I see "All checks have passed" and the log looks good...

**Comment by stefanv** [comment]
Perhaps Drone does run to completion but fails to store the entire output log?

**Comment by acrellin** [comment]
You must be in that parallel universe Stefan slipped into yesterday...

**Comment by bnaul** [comment]
One commit still had `mltsp.datasets` in `conf.py` and I amended that one; another time the build failed because of an internet outage? `ConnectionError: HTTPConnectionPool(host='travis-wheels.scikit-image.org', port=80): Max retries exceeded.` I re-built and it looks good.

**Comment by stefanv** [comment]
I checked the build on your branch, and Drone was happy with that one, so merging.

**Comment by bnaul** [comment]
Nooooooo https://readthedocs.org/projects/mltsp/builds/3578426/

**Comment by bnaul** [comment]
Why is it building JSON anyway?? Looks like the HTML is fine at least :weary:

**Comment by stefanv** [comment]
No, I'm not sure it is fine.

**Comment by bnaul** [comment]
?

**Comment by stefanv** [comment]
The HTML did not get uploaded, so how do we check?

**Comment by bnaul** [comment]
The HTML on readthedocs right now says "Revision da86ddd5." which is this one, so I think it did get updated. Could be wrong though. Regardless the new PR should fix it.

**Comment by stefanv** [comment]
Last successful build was 1 day 1 hr ago.  Not sure which one that was!

**Pull request merged**